assets/js/modules/analytics/components/settings/OptionalSettingsView.js

```diff
@@ -48,23 +55,36 @@ export default function OptionalSettingsView() {
 	const trackingDisabled = useSelect(
 		( select ) => select( MODULES_ANALYTICS ).getTrackingDisabled() || []
 	);
+	const useGA4Snippet = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getUseSnippet()
+	);
 
 	const ampMode = useSelect( ( select ) => select( CORE_SITE ).getAMPMode() );
 
-	const showIPAnonymizationSettings = useSnippet && ampMode !== 'primary';
+	const showIPAnonymizationSettings =
+		useSnippet &&
+		ampMode !== 'primary' &&
+		( ! ga4ReportingEnabled || isAnalyticsConnected );
```

Comment:
As mentioned above, `isAnalyticsConnected` here will always be true if either Analytics is connected since it implies `ga4ReportingEnabled` is enabled. This all seems to work fine, but thought I'd mention.

Reply:
Thanks for pointing this out. It looks like things are working because the `useSnippet` setting gets set to `false` when we toggle UA off, so `isAnalyticsConnected` isn't being checked. However, to be on the safe side I've updated this to use a more correct check for the Analytics connected status, as described above, as I think it's worth keeping the condition for the sake of being explicit and also in case the `useSnippet` logic changes or there's an alternative execution path we haven't spotted where `useSnippet` does retain its `true` value.